### PR TITLE
Person, student, teacher classes

### PR DESCRIPTION
**Pull request by anagudelogu**
# In this Milestone I:

- [x]  Created class `Person` with the following:
  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`, `age`, and `parent_permission` as parameters. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and false otherwise.
  - Public method `can_use_services?` returns `true` if a person `is of age` or if they have permission from parents.

 - [x] Created class `Student` with the following:
   - Inherits from `Person`.
   - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
   - Method `play_hooky` that returns `"¯\(ツ)/¯"`.

 - [x] Created class `Teacher` with the following:
   - Inherits from `Person`.
   - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
   - Override `can_use_services?` so it always returns `true`.

- [x] No linter errors.
- [x] Documented the project in a professional way.

I would appreciate any optional comments and feedback about the code in general, this helps me improve my skills and write code that's clean and follows professional standards. 🤓

### CODE REVIEW

File: lib/person.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
```
Comment (prolajumokeoni) [inline lib/person.rb:5]: - [ ] Kindly add  `  @id = Random.rand(1..1000)`   to your instance variables defined 👍🏿

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @anagudelogu, 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Git flow was correctly used




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: lib/person.rb (hunk 1 above)
Comment (anagudelogu) [reply lib/person.rb:5]: Hi @prolajumokeoni! Thank you for your feedback. I'm doing just that inside my initialize method, isn't that okay?

**Review by Davidosky007 — APPROVED** [review]
Hi @anagudelogu 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written readme file 👍 
* Good looking PR description 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**